crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -474,6 +475,29 @@ declare_lint! {
     }
 }
 
+declare_lint! {
+    /// ## What it does
+    /// Checks for calls to an overloaded function that do not match any of the overloads.
+    ///
+    /// ## Why is this bad?
+    /// Failing to provide the correct arguments to one of the overloads will raise a `TypeError`
+    /// at runtime.
```

Comment:
Minor point: it's not necessarily true that this will raise a `TypeError`. Calling a function with arguments of the wrong type can result in any kind of error, or none at all.